WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -191,7 +185,7 @@ extension PointOfSaleAggregateModel {
 
     @MainActor
     func collectPayment() async {
-        guard let order else {
+        guard let order = orderController.order else {
```

Comment:
Yes, it should be, but I don't think I have time during this project. The deprecation comment is to prevent us from using that property elsewhere, it's only exposed for this use.

We need it here because the card payment service isn't stateless, because it relies on having the order after the initial `collectPayment` call is made. It needs the order for retries after a payment error, and to automatically collect payment when the reader connects on the totals view.

It would be possible to pass this state along through the flow, but it would be quite a big job so I've left it out of scope for now.